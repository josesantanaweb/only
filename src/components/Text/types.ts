export type tagType = "p" | "span";

export interface TextProps {
  /** Sets class text */
  className?: string;
  /** Sets tag text */
  tag?: tagType;
  /** Sets type text */
  type: "body-small" | "body-medium" | "body-large" | "button";
  /** Sets as text */
  as?: tagType;
  /** Sets color text */
  variant?: "primary" | "secondary" | "tertiary" | "white" | "black" | "grey";
  /** Sets text-transform text */
  uppercase?: boolean;
}
