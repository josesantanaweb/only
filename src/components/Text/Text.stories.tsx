import React from "react";

import { ComponentStory, ComponentMeta } from "@storybook/react";

import Text from "./Text";

export default {
  title: "Text",
  component: Text,
  argTypes: {
    type: {
      control: {
        type: "select",
        options: ["body-small", "body-medium", "body-large", "button"],
      },
    },
  },
} as ComponentMeta<typeof Text>;

const Template: ComponentStory<typeof Text> = (args) => <Text {...args} />;

export const Texto = Template.bind({});
Texto.args = {
  tag: "p",
  type: "body-medium",
  children: "Text",
  variant: "grey",
};
