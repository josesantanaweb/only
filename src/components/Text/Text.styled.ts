import styled, { css } from "styled-components";
import { TextProps } from "./types";

export const TextStyled = styled.p<TextProps>(
  ({ theme, uppercase, type, variant }) => {
    const uppercaseMixin = css`
      text-transform: ${uppercase ? "uppercase" : "capitalize"};
    `;

    const variantMixin = () => {
      switch (variant) {
        case "primary":
          return css`
            color: ${theme.colors.primary};
          `;
        case "secondary":
          return css`
            color: ${theme.colors.secondary};
          `;
        case "black":
          return css`
            color: ${theme.colors.black};
          `;
        case "white":
          return css`
            color: ${theme.colors.white};
          `;
        case "grey":
          return css`
            color: ${theme.colors.grey};
          `;
        default:
          return css`
            color: ${theme.colors.tertiary};
          `;
      }
    };

    const typeMixin = () => {
      switch (type) {
        case "body-small":
          return css`
            font-size: ${theme.font.sizes.bodySmall};
            line-height: 16px;
            font-weight: ${theme.font.regular};
          `;
        case "body-medium":
          return css`
            font-size: ${theme.font.sizes.bodyMedium};
            line-height: 20px;
            font-weight: ${theme.font.regular};
          `;
        case "body-large":
          return css`
            font-size: ${theme.font.sizes.bodyLarge};
            line-height: 27px;
            font-weight: ${theme.font.regular};
          `;
        case "button":
          return css`
            font-size: ${theme.font.sizes.button};
            line-height: 21px;
            font-weight: ${theme.font.regular};
          `;
        default:
          return css`
            font-size: 14px;
            line-height: 21px;
            font-weight: ${theme.font.regular};
          `;
      }
    };

    return css`
      display: inline-block;
      margin: 0;
      font-weight: 500;
      padding: 0;
      line-height: 1;
      ${typeMixin};
      ${variantMixin};
      ${uppercaseMixin};
    `;
  }
);

export default TextStyled;
