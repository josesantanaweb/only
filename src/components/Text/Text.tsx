import React, { FC } from "react";
import TextStyled from "./Text.styled";
import { TextProps } from "./types";

const Text: FC<TextProps> = ({
  children,
  tag,
  type,
  className,
  uppercase = false,
  variant,
}) => (
  <TextStyled
    as={tag}
    className={className}
    type={type}
    uppercase={uppercase}
    variant={variant}
  >
    {children}
  </TextStyled>
);

export default Text;
