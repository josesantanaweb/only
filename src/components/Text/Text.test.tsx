import React from "react";
import { render } from "../../test";
import Text from "./Text";

describe("Text", () => {
  it("renders the text correctly", () => {
    const { container } = render(<Text type="body-medium">Texto</Text>);
    expect(container).toMatchSnapshot();
  });

  it("should render with all the props", () => {
    const { container } = render(
      <Text variant="grey" tag="p" type="body-large">
        Texto
      </Text>
    );
    expect(container).toMatchSnapshot();
  });
});
