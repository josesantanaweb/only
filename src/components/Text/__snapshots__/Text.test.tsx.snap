// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text renders the text correctly 1`] = `
<div>
  <p
    class="sc-bdnxRM cALYuU"
    type="body-medium"
  >
    Texto
  </p>
</div>
`;

exports[`Text should render with all the props 1`] = `
<div>
  <p
    class="sc-bdnxRM aLiKR"
    type="body-large"
  >
    Texto
  </p>
</div>
`;
